lib/bibliothecary/runner.rb

```diff
@@ -42,6 +42,22 @@ def package_managers
       Bibliothecary::Parsers.constants.map{|c| Bibliothecary::Parsers.const_get(c) }.sort_by{|c| c.to_s.downcase }
     end
 
+    def load_file_info_list_from_files(files)
+      # Parses an array of format [{file_path: "", contents: ""},] to match
+      #  on both filename matches, and one content_match patterns.
+      file_list = []
+
+      files.each do |file|
+        info = FileInfo.new(nil, file[:file_path], file[:contents])
```

Comment:
This is the MAIN change here, adding contents to come in here if we have the contents.

in the FileInfo stuff its read from a temp location, so setting here does all the checks properly.

Reply:
everythgin else is just new methods to explicitly call this out
(vs just checking "is array of strings, vs is array of objects" shenanigans)